### PR TITLE
media-sound/setbfree: new package

### PR DESCRIPTION
**Pull request by marcan**
Package-Manager: Portage-2.3.6, Repoman-2.3.2

### CODE REVIEW

File: media-sound/setbfree/setbfree-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+inherit toolchain-funcs multilib
+
+DESCRIPTION="MIDI controlled DSP tonewheel organ"
+HOMEPAGE="http://setbfree.org"
+
+if [ "${PV}" = "9999" ]; then
+	inherit subversion git-r3
```
Comment (simonvanderveldt) [inline media-sound/setbfree/setbfree-9999.ebuild]: @marcan I don't think subversion is correct here :)

File: media-sound/setbfree/setbfree-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+else
+	SRC_URI="https://github.com/pantherb/setBfree/releases/download/v${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-2+"
+SLOT="0"
+IUSE="convolution"
+
+RDEPEND="media-sound/jack-audio-connection-kit
```
Comment (simonvanderveldt) [inline media-sound/setbfree/setbfree-9999.ebuild]: Unless setbfree only works with jack1 this should be `virtual/jack`

File: media-sound/setbfree/setbfree-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	>=media-libs/alsa-lib-1.0.0:=
+	media-libs/liblo:=
+	media-libs/lv2
```
Comment (simonvanderveldt) [inline media-sound/setbfree/setbfree-9999.ebuild]: Is it not possible to choose what to build (standalone application or lv2 plugin)/

File: media-sound/setbfree/setbfree-9999.ebuild [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	convolution? ( media-libs/libsndfile:=
+		>=media-libs/zita-convolver-3.1.0:= )
+	media-fonts/dejavu"
```
Comment (simonvanderveldt) [inline media-sound/setbfree/setbfree-9999.ebuild]: I believe this shoud be `virtual/ttf-fonts-1-r1`?
Or is it hardcoded to dejavu?

Comment (marcan) [inline media-sound/setbfree/setbfree-9999.ebuild]: It actually used to be hardcoded to ttf-bitstream-vera, which is on its way out, so I changed it to dejavu (see later in the ebuild for the TTF path). It just uses a hardcoded TTF path, doesn't dynamically find a font :(

File: media-sound/setbfree/setbfree-9999.ebuild (hunk 1 above)
Comment (marcan) [inline media-sound/setbfree/setbfree-9999.ebuild]: Err. Yeah. Fixed.

File: media-sound/setbfree/setbfree-9999.ebuild (hunk 2 above)
Comment (marcan) [inline media-sound/setbfree/setbfree-9999.ebuild]: Fixed.

File: media-sound/setbfree/setbfree-9999.ebuild (hunk 3 above)
Comment (marcan) [inline media-sound/setbfree/setbfree-9999.ebuild]: It's a mess of automagic junk in a Makefile with no easy way to configure what goes in and what doesn't. As far as I can tell, the UI (even in standalone mode) depends on LV2. You can build the synth core without LV2 and the UI, but there's no way to force that (well, unless you do something ugly like set FONTFILE to a non-existent file to force the UI test to fail). And the synth core by itself is of limited use unless you already have all the presets you need (headless mode?) or are happy using MIDI to control everything.

**Comment by stefantalpalaru** [comment]
The newest version is https://github.com/pantherb/setBfree/archive/v0.8.4.tar.gz .

**Comment by marcan** [comment]
Welp, looks like upstream isn't using GitHub releases consistently (0.8.0 was the last available *release*).

Bumped to 0.8.4.

File: media-sound/setbfree/setbfree-9999.ebuild (hunk 4 above)
Comment (simonvanderveldt) [inline media-sound/setbfree/setbfree-9999.ebuild]: Seems like it still defaults to ttf-bitstream-vera https://github.com/pantherb/setBfree/blob/9c04792f1c16725eb978527add672eae2f938f26/common.mak#L6 :(

But I understand, not much we can then :(

**Pull request merged**